R/harmonise_method.R

```diff
@@ -89,46 +94,47 @@ setMethod("overlapVariants", "DataSet", function(object, strand = "forward") {
   return(object)
 })
 
+
+
 # Set and get methods for harmonise data
 
 #' Harmonise the alleles and effects between two summary sets
-#' @param object The DataSet object
+#' 
+#' @param object The gwasglue2 DataSet object
 #' @param action Level of strictness in dealing with SNPs.
 #' * `action = 1`: Assume all alleles are coded on the forward strand, i.e. do not attempt to flip alleles
 #' * `action = 2`: Try to infer positive strand alleles, using allele frequencies for palindromes (default, conservative);
 #' * `action = 3`: Correct strand for non-palindromic SNPs, and drop all palindromic SNPs from the analysis (more conservative).
 #' @param tolerance Tolerance value.
-#' @param strand dna strand orientation  (Default "forward", other option "reverse")
-#' 
-setGeneric("harmoniseData", function(object, tolerance, action, strand) standardGeneric("harmoniseData"))
-setMethod( "harmoniseData", "DataSet", function(object, tolerance = 0.08, action = 2, strand = "forward"){
+#' @return The gwasglue2 DataSet object harmonised
+#' @docType methods
+#' @rdname harmoniseData-methods
+setGeneric("harmoniseData", function(object, tolerance, action) standardGeneric("harmoniseData"))
+#' @rdname harmoniseData-methods
+setMethod( "harmoniseData", "DataSet", function(object, tolerance = 0.08, action = 2){
```

Comment:
Before harmonising, the `overlapVariants()` function has a `strand` argument. If "forward" looks for common variants using `variantid`. If "reverse" it removes multiallelics and uses "rsid". After this step the pipeline always uses `variantid`

Reply:
I think there will never be a dataset on the reverse strand. There will either be only forward strand alleles, or very occasionally we won't know what the strand is.

I don't think you need the reverse vs forward when you have actions 1,2,3? Or maybe I misunderstand.

But my point about `action=1` is that we don't need to run the stuff in here https://github.com/MRCIEU/gwasglue2/blob/cfd6d08d159f0d02c6a5c369d3c77f6321d7fb0c/R/harmonise_method.R#L114 because when the alleles are standardised alphabetically then you don't need to do the line-by-line comparison. Only need to subset to have variantids that are in common